src/main/kotlin/org/neo4j/graphql/Translator.kt

```diff
@@ -397,7 +459,26 @@ object SchemaBuilder {
         """
         typeDefinitionRegistry.merge(schemaParser.parse(directivesSdl))
 
-        val augmentations = typeDefinitionRegistry.types().values.filterIsInstance<ObjectTypeDefinition>().map { augmentedSchema(ctx, it) }
+        val objectTypeDefinitions = typeDefinitionRegistry.types().values.filterIsInstance<ObjectTypeDefinition>()
+        val nodeAugmentations = objectTypeDefinitions.map { augmentedSchema(ctx, it) }
+        val relAugmentations = objectTypeDefinitions.flatMap { source ->
+            val fields = source.fieldDefinitions.filter { !it.type.isScalar() && it.directives.isEmpty() }
+            if (fields.isEmpty()) {
+                emptyList<Augmentation>()
+            }
+            val augmentations = fields.mapNotNull { targetField ->
+                val target = objectTypeDefinitions
```

Comment:
why don't we just check if the target type is an Object type?
Need to use .inner() as it can be a non-null or list field of the target type.

Reply:
Ok